event-loop.go

```diff
@@ -46,17 +49,10 @@ func newEventLoop(postEmptyEvent func(), onExpiredTask func(*embedder.FlutterTas
 	}
 }
 
-// RunOnCurrentThread FlutterDocs:
-//   May be called from any thread. Should return true if tasks posted on the
-//   calling thread will be run on that same thread.
-//
-// The functions PostTask and onExpiredTask should be called from the same
-// thread, this is ensured if the creation of the event loop (through
-// `newEventLoop`) and the PostTask callback (through
-// `a.engine.TaskRunnerPostTask = eventLoop.PostTask`) are done on a calling
-// goroutine which always execute in that thread (`runtime.LockOSThread()`).
```

Comment:
This assumption was wrong.

Reply:
Hmm that's interesting... I'm not sure I understand which behaved different than expected..?

Reply:
The call to `RunOnCurrentThread` is done by the engine, on an internally managed thread.
The assumption that `runtime.LockOSThread()` would also lock on which thread this Cb is executed is wrong.

When the task_runner was only wired for `platform_task_runner`, the engine was always piking the correct thread to handle the cb.
But when adding `render_task_runner` to the same task_runner, the engine first try to run the GLFW cb (context current, swapbuffer,.. stuff defined in application.go) on an non-main thread, it's only on the seconds try (after this `RunOnCurrentThread` returned false once), that the engine select the correct thread to run those GLFW Cbs.